---
title: "Visualization of SexFindR Steps"
author: "Shannon EK Joslin"
date: "2023-03-24"
output: 
  html_document:
    toc: true
    toc_depth: 4
---

```{r setup, eval=TRUE, include=FALSE}
knitr::opts_chunk$set(warning=FALSE, message=FALSE, echo=FALSE)
library(tidyverse)
library(ggpubr)
library(gridExtra)
setwd("/Users/miocene/Desktop/git_repos/dissertation/reanalyzed/sex_marker/SexFindR")
wd <- getwd()
```

## _Step 1: difcover_

### TKTKT WORKING IN `tmp4quickknit.Rmd`




## Step 2: SNPdensity
### Female Reference Genome
### Male Reference Genome

## Step 2: GWAS & Fst
### Female Reference Genome
#### PCA
##### All individuals

```{r pca_femaleRef_all, echo=FALSE}
### FEMALE REFERENCE ###
# read in data
pca_f <- read_table2("./step2-gwas/pca/femaleRef_pca.eigenvec", col_names = FALSE)
eigenval_f <- scan("./step2-gwas/pca/femaleRef_pca.eigenval")
# sort out the pca data
# remove nuisance column
pca_f <- pca_f[,-1]
# set names
names(pca_f)[1] <- "ind"
names(pca_f)[2:ncol(pca_f)] <- paste0("PC", 1:(ncol(pca_f)-1))
# sort sex
sex_f <- rep(NA, length(pca_f$ind))
sex_f[grep("_F_", pca_f$ind)] <- "F"
sex_f[grep("_M_", pca_f$ind)] <- "M"
# remake data.frame
pca_f <- as.tibble(data.frame(pca_f, sex_f))
```

```{r pca_femaleRef_all_pve, echo=FALSE}
# first convert to percentage variance explained
pve_f <- data.frame(PC = 1:20, pve_f = eigenval_f/sum(eigenval_f)*100)
# make plot
a_f <- ggplot(pve_f, aes(PC, pve_f)) +
  geom_bar(stat = "identity") +
  ylab("Percentage variance explained") +
  theme_light()
a_f
# calculate the cumulative sum of the percentage variance explained 
cumsum(pve_f$pve_f)
```

```{r pca_femaleRef_all_scatter, echo=FALSE}
# plot pca
b_f <- ggplot(pca_f, aes(PC1, PC2, col = sex_f)) +
  geom_point(size = 3) +
  scale_colour_manual(values = c("darkred", "darkblue")) +
  coord_equal() +
  xlab(paste0("PC1 (", signif(pve_f$pve_f[1], 3), "%)")) +
  ylab(paste0("PC2 (", signif(pve_f$pve_f[2], 3), "%)"))
print("full dataset - female ref")
b_f
```

```{r pca_femaleRef_all_scatter_zoom, echo=FALSE}
# zoom plot
pca_subset_f <- pca_f %>% filter(PC1 < 0.25, PC2 > -0.1, PC2 < 0.25)
c_f <- ggplot(pca_subset_f, aes(PC1, PC2, col = sex_f)) +
  geom_point(size = 3) +
  scale_colour_manual(values = c("darkred", "darkblue")) +
  coord_equal()
print("full dataset - female ref - zoom")
c_f
# combined plot
b_f <- b_f + theme(legend.position = "none")
grid.arrange(b_f, c_f, ncol=2)
#grid.arrange(b_f, c_f, ncol=2, top=textGrob("Female Reference Assembly \n Principal Component Analysis of All Indivduals"))
# save plot
b_f_big <- b_f + theme(text=element_text(size=24))
c_f_big <- c_f + theme(text=element_text(size=24))
```

```{r pca_femaleRef_all_save}
png(filename="visualizations/Figure_Step2-GWAS_femaleRef_PCA_all_individuals.png", width=1020, height=570)
print("full dataset - female ref")
grid.arrange(b_f_big, c_f_big, ncol=2)
dev.off()

#pca_zoom <- pca %>% filter(PC1 == (-0.10000:0.10000) & PC2 == (-0.1:0.1))
#ggplot(pca, aes(PC1, PC2, col = sex)) +
#  geom_point(size = 3) +
#  scale_colour_manual(values = c("darkred", "darkblue")) +
#  coord_equal() +
#  xlab(paste0("PC1 (", signif(pve$pve[1], 3), "%)")) +
#  ylab(paste0("PC2 (", signif(pve$pve[2], 3), "%)")) +
#  theme(text=element_text(size=20), axis.text.x=element_text(angle=45, hjust=1)) +
#  #geom_mark_rect(data=pca_zoom, expand = unit(1.5, "mm"), fill="lightgreen") +
#  facet_zoom(xlim=c(-0.1,0.1), ylim=c(-0.1, 0.1))
```

```{r pca_femaleRef_all_outlier_list, echo=TRUE}
# grab outliers
pca_outliers_f <- pca_f %>% filter(PC1 > 0.25 | PC2 < -0.1 | PC2 > 0.25)
print(pca_outliers_f$ind)
```

##### No Outliers

```{r pca_femaleRef_no_outliers, echo=FALSE}
### FEMALE REFERENCE ###
# read in data
pca_f_no <- read_table2("./step2-gwas/output_no_outliers/femaleRef_no_outliers.eigenvec", col_names = FALSE)
eigenval_f_no <- scan("./step2-gwas/output_no_outliers/femaleRef_no_outliers.eigenval")
# sort out the pca data
# remove nuisance column 
pca_f_no <- pca_f_no[,-1]
# set names
names(pca_f_no)[1] <- "ind"
names(pca_f_no)[2:ncol(pca_f_no)] <- paste0("PC", 1:(ncol(pca_f_no)-1))
# sort sex
sex_f_no <- rep(NA, length(pca_f_no$ind))
sex_f_no[grep("_F_", pca_f_no$ind)] <- "F"
sex_f_no[grep("_M_", pca_f_no$ind)] <- "M"
# remake data.frame
pca_f_no <- as.tibble(data.frame(pca_f_no, sex_f_no))
```

```{r pca_femaleRef_no_outliers_pve, echo=FALSE}
# first convert to percentage variance explained
pve_f_no <- data.frame(PC=1:20, pve_f_no=eigenval_f_no/sum(eigenval_f_no)*100)
# make plot
a_f_no <- ggplot(pve_f_no, aes(PC, pve_f_no)) +
  geom_bar(stat = "identity") +
  ylab("Percentage variance explained") +
  theme_light()
# make plot
a_f_no <- ggplot(pve_f_no, aes(PC, pve_f_no)) + geom_bar(stat = "identity")
a_f_no + ylab("Percentage variance explained")
# calculate the cumulative sum of the percentage variance explained
cumsum(pve_f_no$pve_f_no)
```

```{r pca_femaleRef_no_outliers_scatter, echo=FALSE}
# plot pca with big axis
b_f_no <- ggplot(pca_f_no, aes(PC1, PC2, col = sex_f_no)) +
  geom_point(size = 3) +
  scale_colour_manual(values = c("orangered", "steelblue3")) +
  coord_equal() +
  scale_y_continuous(limits=c(-0.5, 0.5)) +
  scale_x_continuous(limits=c(-0.5, 0.5)) +
  xlab(paste0("PC1 (", signif(pve_f_no$pve_f_no[1], 3), "%)")) +
  ylab(paste0("PC2 (", signif(pve_f_no$pve_f_no[2], 3), "%)"))
print("no outliers - female ref")
b_f_no
```

```{r pca_femaleRef_no_outliers_scatter_zoom, echo=FALSE}
# plot pca
c_f_no <- ggplot(pca_f_no, aes(PC1, PC2, col = sex_f_no)) +
  geom_point(size = 3) +
  scale_y_continuous(limits=c(-0.5, 0.5)) +
  scale_x_continuous(limits=c(-0.4, 0.2)) +
  scale_colour_manual("sex", values = c("orangered", "steelblue3")) +
  coord_equal()
print("no outliers - female ref - zoom")
c_f_no
```

```{r pca_femaleRef_no_outliers_facet_plots, echo=FALSE}
# combined plot (all individuals vs no outliers)
b_f_no <- b_f_no + theme(legend.position = "none")
print("no outliers - female ref")
grid.arrange(b_f, c_f, b_f_no, c_f_no, ncol=2, nrow=2)
#grid.arrange(b_f, c_f, b_f_no, c_f_no, ncol=2, nrow=2, top=textGrob("Female Reference Assembly \n Principal Component Analysis with Outliers"))
# save plot
b_f_no_big <- b_f_no + theme(text=element_text(size=24))
c_f_no_big <- c_f_no + theme(text=element_text(size=24))
```

```{r pca_femaleRef_no_outliers_save}
png(filename="visualizations/Figure_Step2-GWAS_femaleRef_PCA_no_outliers.png", width=1220, height=570)
print("female ref\n row1=full dataset, row2=filtered")
grid.arrange(b_f_big, c_f_big, b_f_no_big, c_f_no_big, ncol=2, nrow=2)
dev.off()
```

#### GWAS
##### All Individuals
##### No Outliers

#### Fst
##### All Individuals
```{r fst_femaleRef_all, echo=FALSE}
Fst_F <- read_tsv("./step2-fst/biallelic_fst_femaleRef.weir.fst") %>% replace_na(list(WEIR_AND_COCKERHAM_FST=0)) %>% rename(scaf = CHROM, base=POS) %>% mutate(base=as.numeric(base)) %>% filter(WEIR_AND_COCKERHAM_FST >=0) %>% replace_na(list(WEIR_AND_COCKERHAM_FST=0))
scaffold_lengths_F <- read_tsv("./step0-variantcalling/femaleRef/CHRR_integrated.fa.fai", col_names = c("scaf","length")) %>% filter(grepl("lg",scaf))
cutoff1_F <- round(nrow(Fst_F)*0.05)
Fst_F_sort_cut1 <- head(Fst_F %>% arrange(-WEIR_AND_COCKERHAM_FST), n=cutoff1_F)
min(Fst_F_sort_cut1$WEIR_AND_COCKERHAM_FST) # lowest value there is 0.0737052 (5% cutoff)
min_five_cutoff_F <- min(Fst_F_sort_cut1$WEIR_AND_COCKERHAM_FST)
cutoff2_F <- round(nrow(Fst_F)*0.01)
Fst_F_sort_cut2 <- head(Fst_F %>% arrange(-WEIR_AND_COCKERHAM_FST), n=cutoff2_F)
min(Fst_F_sort_cut2$`WEIR_AND_COCKERHAM_FST`) # lowest value there is 0.243856 (1% cutoff)
min_one_cutoff_F <- min(Fst_F_sort_cut2$WEIR_AND_COCKERHAM_FST)
table(Fst_F_sort_cut2$scaf)
quantile(Fst_F$WEIR_AND_COCKERHAM_FST, c(0.95, 0.99), na.rm = T)
mean(Fst_F$WEIR_AND_COCKERHAM_FST)
Fst_female <- left_join(scaffold_lengths_F,Fst_F) %>% filter(WEIR_AND_COCKERHAM_FST>=0)

#for x axis, we want cumulative bases for each position in the genome for a continuous axis
nCHR_F <- length(unique(Fst_female$scaf))
Fst_female$BPcum <- 0
sf <- 0
nbp_F <- c()
for (i in unique(Fst_female$scaf)){
  nbp_F[i] <- max(Fst_female[Fst_female$scaf == i,]$base)
  Fst_female[Fst_female$scaf == i,"BPcum"] <- Fst_female[Fst_female$scaf == i,"base"] + sf
  sf <- sf + nbp_F[i]
}
axis.set_F <- Fst_female %>%
  group_by(scaf) %>%
  summarize(center = (max(BPcum) + min(BPcum)) / 2)
Fst_female %>% ggplot(aes(x=BPcum, y=WEIR_AND_COCKERHAM_FST, color=as.factor(scaf))) +
  geom_point(alpha = 0.7) +
  scale_x_continuous(label = axis.set_F$scaf, breaks = axis.set_F$center) +
  theme(axis.text.x = element_text(angle = 90), legend.position="none") +
  labs(x="Chomosome", y="Weir & Cockerham Fst", color="") +
  #guides(color = guide_legend(ncol=2, byrow=F)) +
  geom_hline(yintercept = min_five_cutoff_F, linetype="dotted", color = "black", size=0.75) +
  geom_hline(yintercept = min_one_cutoff_F, linetype="dotted", color = "black", size=1.5)  +
  #geom_vline(xintercept = as.integer(sf), linetype="dotted", color="red", size=1) +
  labs(title="Fst Female Reference - no filters\n (5% and 1% cutoff as dotted lines)")
ggsave("visualizations/Figure_Step2-Fst_femaleRef_manhattan_with95_99_noMissing_all.pdf")
```

##### No Outliers
```{r fst_femaleRef_no_outliers_lg, echo=FALSE}
Fst_F <- read_tsv("./step2-fst/subset_analyses/femaleRef/femaleRef_no_outliers_pruned_fst.weir.fst") %>% replace_na(list(WEIR_AND_COCKERHAM_FST=0)) %>% rename(scaf = CHROM, base=POS) %>% mutate(base=as.numeric(base)) %>% filter(WEIR_AND_COCKERHAM_FST >=0) %>% replace_na(list(WEIR_AND_COCKERHAM_FST=0))
scaffold_lengths_F <- read_tsv("./step0-variantcalling/femaleRef/CHRR_integrated.fa.fai", col_names = c("scaf","length")) %>% filter(grepl("lg",scaf))
cutoff1_F <- round(nrow(Fst_F)*0.05)
Fst_F_sort_cut1 <- head(Fst_F %>% arrange(-WEIR_AND_COCKERHAM_FST), n=cutoff1_F)
min(Fst_F_sort_cut1$WEIR_AND_COCKERHAM_FST) # lowest value there is 0.118254  (5% cutoff)
min_five_cutoff_F <- min(Fst_F_sort_cut1$WEIR_AND_COCKERHAM_FST)
cutoff2_F <- round(nrow(Fst_F)*0.01)
Fst_F_sort_cut2 <- head(Fst_F %>% arrange(-WEIR_AND_COCKERHAM_FST), n=cutoff2_F)
min(Fst_F_sort_cut2$`WEIR_AND_COCKERHAM_FST`) # lowest value there is 0.193863 (1% cutoff)
min_one_cutoff_F <- min(Fst_F_sort_cut2$WEIR_AND_COCKERHAM_FST)
table(Fst_F_sort_cut2$scaf)
quantile(Fst_F$WEIR_AND_COCKERHAM_FST, c(0.95, 0.99), na.rm = T)
mean(Fst_F$WEIR_AND_COCKERHAM_FST) # 0.03776015
Fst_female <- left_join(scaffold_lengths_F,Fst_F) %>% filter(WEIR_AND_COCKERHAM_FST>=0)

#for x axis, we want cumulative bases for each position in the genome for a continuous axis
nCHR_F <- length(unique(Fst_female$scaf))
Fst_female$BPcum <- 0
sf <- 0
nbp_F <- c()
for (i in unique(Fst_female$scaf)){
  nbp_F[i] <- max(Fst_female[Fst_female$scaf == i,]$base)
  Fst_female[Fst_female$scaf == i,"BPcum"] <- Fst_female[Fst_female$scaf == i,"base"] + sf
  sf <- sf + nbp_F[i]
}
axis.set_F <- Fst_female %>%
  group_by(scaf) %>%
  summarize(center = (max(BPcum) + min(BPcum)) / 2)
Fst_female %>% ggplot(aes(x=BPcum, y=WEIR_AND_COCKERHAM_FST, color=as.factor(scaf))) +
  geom_point(alpha=0.75) +
  scale_x_continuous(label=axis.set_F$scaf, breaks=axis.set_F$center) +
  theme(axis.text.x=element_text(angle=90)) +
  labs(x="Chromosome", y="Weir & Cockerham Fst", color="") +
  guides(color=guide_legend(ncol=2, byrow=F)) +
  geom_hline(yintercept=min_five_cutoff_F, linetype="dotted", color = "black", size=0.75) +
  geom_hline(yintercept=min_one_cutoff_F, linetype="dotted", color = "black", size=1.5)  +
  #geom_vline(xintercept=as.integer(sf), linetype="dotted", color="red", size=1) +
  labs(title="Fst Female Reference - filtered dataset\n (5% and 1% cutoff)")
ggsave("visualizations/Figure_Step2-Fst_femaleRef_manhattan_with95_99_noMissing_no_outliers.pdf")
```

```{r fst_femaleRef_no_outliers_scaffold, echo=FALSE}
Fst_F <- read_tsv("./step2-fst/subset_analyses/femaleRef/femaleRef_no_outliers_pruned_fst.weir.fst") %>% replace_na(list(WEIR_AND_COCKERHAM_FST=0)) %>% rename(scaf=CHROM, base=POS) %>% mutate(base=as.numeric(base)) %>% filter(WEIR_AND_COCKERHAM_FST >=0) %>% replace_na(list(WEIR_AND_COCKERHAM_FST=0))
scaffold_lengths_F <- read_tsv("./step0-variantcalling/femaleRef/CHRR_integrated.fa.fai", col_names = c("scaf","length")) %>% filter(grepl("scaffold",scaf))
cutoff1_F <- round(nrow(Fst_F)*0.05)
Fst_F_sort_cut1 <- head(Fst_F %>% arrange(-WEIR_AND_COCKERHAM_FST), n=cutoff1_F)
min(Fst_F_sort_cut1$WEIR_AND_COCKERHAM_FST) # lowest value there is 0.118254  (5% cutoff)
min_five_cutoff_F <- min(Fst_F_sort_cut1$WEIR_AND_COCKERHAM_FST)
cutoff2_F <- round(nrow(Fst_F)*0.01)
Fst_F_sort_cut2 <- head(Fst_F %>% arrange(-WEIR_AND_COCKERHAM_FST), n=cutoff2_F)
min(Fst_F_sort_cut2$`WEIR_AND_COCKERHAM_FST`) # lowest value there is 0.193863 (1% cutoff)
min_one_cutoff_F <- min(Fst_F_sort_cut2$WEIR_AND_COCKERHAM_FST)
table(Fst_F_sort_cut2$scaf)
quantile(Fst_F$WEIR_AND_COCKERHAM_FST, c(0.95, 0.99), na.rm = T)
mean(Fst_F$WEIR_AND_COCKERHAM_FST) # 0.03776015
Fst_female <- left_join(scaffold_lengths_F,Fst_F) %>% filter(WEIR_AND_COCKERHAM_FST>=0)

#for x axis, we want cumulative bases for each position in the genome for a continuous axis
nCHR_F <- length(unique(Fst_female$scaf))
Fst_female$BPcum <- 0
sf <- 0
nbp_F <- c()
for (i in unique(Fst_female$scaf)){
  nbp_F[i] <- max(Fst_female[Fst_female$scaf == i,]$base)
  Fst_female[Fst_female$scaf == i,"BPcum"] <- Fst_female[Fst_female$scaf == i,"base"] + sf
  sf <- sf + nbp_F[i]
}
axis.set_F <- Fst_female %>%
  group_by(scaf) %>%
  summarize(center = (max(BPcum) + min(BPcum)) / 2)
Fst_female %>% ggplot(aes(x=BPcum, y=WEIR_AND_COCKERHAM_FST, color=as.factor(scaf))) +
  geom_point(alpha=0.75) +
  scale_x_continuous(label=axis.set_F$scaf, breaks=axis.set_F$center) +
  theme(axis.text.x=element_text(angle=90), legend.position="none") +
  labs(x="Chromosome", y="Weir & Cockerham Fst", color="") +
  #guides(color=guide_legend(ncol=3, byrow=F), ) +
  geom_hline(yintercept=min_five_cutoff_F, linetype="dotted", color = "black", size=0.75) +
  geom_hline(yintercept=min_one_cutoff_F, linetype="dotted", color = "black", size=1.5)  +
  #geom_vline(xintercept=as.integer(sf), linetype="dotted", color="red", size=1) +
  labs(title="Fst Female Reference - filtered dataset - scaffolds\n (5% and 1% cutoff)")
ggsave("visualizations/Figure_Step2-Fst_femaleRef_manhattan_with95_99_noMissing_no_outliers_scaffolds.pdf")
```


### Male Reference Genome
#### PCA
##### All Individuals

```{r pca_maleRef_all, echo=FALSE}
### MALE REFERENCE ###
# read in data
pca_m <- read_table2("./step2-gwas/pca/maleRef_pca.eigenvec", col_names = FALSE)
eigenval_m <- scan("./step2-gwas/pca/maleRef_pca.eigenval")
# sort out the pca data
# remove nuisance column
pca_m <- pca_m[,-1]
# set names
names(pca_m)[1] <- "ind"
names(pca_m)[2:ncol(pca_m)] <- paste0("PC", 1:(ncol(pca_m)-1))
# sort sex
sex_m <- rep(NA, length(pca_m$ind))
sex_m[grep("_F_", pca_m$ind)] <- "F"
sex_m[grep("_M_", pca_m$ind)] <- "M"
# remake data.frame
pca_m <- as.tibble(data.frame(pca_m, sex_m))
```

```{r pca_maleRef_all_pve, echo=FALSE}
# first convert to percentage variance explained
pve_m <- data.frame(PC = 1:20, pve = eigenval_m/sum(eigenval_m)*100)
# make plot
a_m <- ggplot(pve_m, aes(PC, pve)) + geom_bar(stat = "identity")
a_m + ylab("Percentage variance explained")
# calculate the cumulative sum of the percentage variance explained
cumsum(pve_m$pve)
```

```{r pca_maleRef_all_scatter, echo=FALSE}
# plot pca
b_m <- ggplot(pca_m, aes(PC1, PC2, col = sex_m)) + geom_point(size = 3) +
  scale_colour_manual(values = c("darkred", "darkblue")) +
  coord_equal() +
  xlab(paste0("PC1 (", signif(pve_m$pve[1], 3), "%)")) + 
  ylab(paste0("PC2 (", signif(pve_m$pve[2], 3), "%)"))
print("full dataset - male ref")
b_m
```

```{r pca_maleRef_all_scatter_zoom, echo=FALSE}
# zoom plot
pca_subset_m <- pca_m %>% filter(PC1 > -0.25, PC2 > -0.25)
c_m <- ggplot(pca_subset_m, aes(PC1, PC2, col = sex_m)) + geom_point(size = 3) + 
  scale_colour_manual(values = c("darkred", "darkblue")) + 
  coord_equal()
print("full dataset - male ref - zoom")
c_m
```

```{r pca_maleRef_all_facet_plot, echo=FALSE}
# combined plot
b_m <- b_m + theme(legend.position = "none")
grid.arrange(b_m, c_m, ncol=2)
#grid.arrange(b_m, c_m, ncol=2, top=textGrob("Male Reference Assembly \n Principal Component Analysis of All Indivduals"))
# save plot with bigger text and one legend
b_m_big <- b_m + theme(text=element_text(size=24))
c_m_big <- c_m + theme(text=element_text(size=24))
```

```{r pca_maleRef_all_save}
png(filename="visualizations/Figure_Step2-GWAS_maleRef_PCA_all_individuals.png", width=1020, height=570)
grid.arrange(b_m_big, c_m_big, ncol=2)
dev.off()
```

```{r pca_maleRef_all_outlier_list, echo=TRUE}
# grab outliers
pca_outliers_m <- pca_m %>% filter(PC1 < -0.25 | PC2 < -0.25)
head(pca_outliers_m$ind)
```


##### No Outliers

```{r pca_maleRef_no_outliers, echo=FALSE}
### MALE REFERENCE ###
# read in data
pca_m_no <- read_table2("./step2-gwas/output_no_outliers/maleRef_no_outliers.eigenvec", col_names = FALSE)
eigenval_m_no <- scan("./step2-gwas/output_no_outliers/maleRef_no_outliers.eigenval")
# sort out the pca data
# remove nuisance column 
pca_m_no <- pca_m_no[,-1]
# set names
names(pca_m_no)[1] <- "ind"
names(pca_m_no)[2:ncol(pca_m_no)] <- paste0("PC", 1:(ncol(pca_m_no)-1))
# sort sex
sex_m_no <- rep(NA, length(pca_m_no$ind))
sex_m_no[grep("_F_", pca_m_no$ind)] <- "F"
sex_m_no[grep("_M_", pca_m_no$ind)] <- "M"
# remake data.frame
pca_m_no <- as.tibble(data.frame(pca_m_no, sex_m_no))
```

```{r pca_maleRef_no_outliers_pve, echo=FALSE}
# first convert to percentage variance explained
pve_m_no <- data.frame(PC = 1:20, pve_m_no = eigenval_m_no/sum(eigenval_m_no)*100)
# make plot
a_m_no <- ggplot(pve_m_no, aes(PC, pve_m_no)) +
  geom_bar(stat = "identity") +
  ylab("Percentage variance explained") +
  theme_light()
# make plot
a_m_no <- ggplot(pve_m_no, aes(PC, pve_m_no)) + geom_bar(stat = "identity")
a_m_no + ylab("Percentage variance explained")
# calculate the cumulative sum of the percentage variance explained
cumsum(pve_m_no$pve_m_no)
# calculate the cumulative sum of the percentage variance explained
cumsum(pve_m_no$pve_m_no)
```

```{r pca_maleRef_no_outliers_scatter, echo=FALSE}
# plot pca with big axis
b_m_no <- ggplot(pca_m_no, aes(PC1, PC2, col = sex_m_no)) +
  geom_point(size = 3) +
  scale_colour_manual(values = c("orangered", "steelblue3")) +
  coord_equal() +
  scale_y_continuous(limits=c(-0.5, 0.5)) +
  scale_x_continuous(limits=c(-0.5, 0.5)) +
  xlab(paste0("PC1 (", signif(pve_m_no$pve[1], 3), "%)")) +
  ylab(paste0("PC2 (", signif(pve_m_no$pve[2], 3), "%)"))
print("No Outliers - MaleRef")
b_m_no
```

```{r pca_maleRef_no_outliers_scatter_zoom, echo=FALSE}
# plot pca
c_m_no <- ggplot(pca_m_no, aes(PC1, PC2, col = sex_m_no)) +
  geom_point(size = 3) +
  scale_y_continuous(limits=c(-0.5, 0.5)) +
  scale_x_continuous(limits=c(-0.4, 0.2)) +
  scale_colour_manual("sex", values = c("orangered", "steelblue3")) +
  coord_equal()
print("No Outliers - MaleRef - Zoom")
c_m_no
```

```{r pca_maleRef_no_outliers_facet_plots, echo=FALSE}
# combined plot (all individuals vs no outliers)
b_m_no <- b_m_no + theme(legend.position = "none")
print("no outliers - female ref")
grid.arrange(b_m, c_m, b_m_no, c_m_no, ncol=2, nrow=2)
#grid.arrange(b_m, c_m, b_m_no, c_m_no, ncol=2, nrow=2, top=textGrob("Male Reference Assembly \n Principal Component Analysis with Outliers"))
# save plot
b_m_no_big <- b_m_no + theme(text=element_text(size=24))
c_m_no_big <- c_m_no + theme(text=element_text(size=24))
```

```{r pca_maleRef_no_outliers_save}
png(filename="visualizations/Figure_Step2-GWAS_maleRef_PCA_no_outliers.png", width=1220, height=570)
print("male ref\n row1=full dataset, row2=filtered")
grid.arrange(b_m_big, c_m_big, b_m_no_big, c_m_no_big, ncol=2, nrow=2)
dev.off()
```

#### GWAS
##### All Individuals
##### No Outlier

#### Fst
##### All Individuals

```{r fst_maleRef_all, echo=FALSE}
Fst_M <- read_tsv("./step2-fst/biallelic_fst_maleRef.weir.fst") %>% replace_na(list(WEIR_AND_COCKERHAM_FST=0)) %>% rename(scaf = CHROM, base=POS) %>% mutate(base=as.numeric(base)) %>% filter(WEIR_AND_COCKERHAM_FST >=0) %>% replace_na(list(WEIR_AND_COCKERHAM_FST=0))
scaffold_lengths_M <- read_tsv("./step0-variantcalling/maleRef/CHRR_integrated.fa.fai", col_names = c("scaf","length")) %>% filter(grepl("lg",scaf))
cutoff1_M <- round(nrow(Fst_M)*0.05)
Fst_M_sort_cut1 <- head(Fst_M %>% arrange(-WEIR_AND_COCKERHAM_FST), n=cutoff1_M)
min(Fst_M_sort_cut1$`WEIR_AND_COCKERHAM_FST`) # lowest value there is 0.074878 (5% cutoff)
min_five_cutoff_M <- min(Fst_M_sort_cut1$`WEIR_AND_COCKERHAM_FST`)
cutoff2_M <- round(nrow(Fst_M)*0.01)
Fst_M_sort_cut2 <- head(Fst_M %>% arrange(-WEIR_AND_COCKERHAM_FST), n=cutoff2_M)
min(Fst_M_sort_cut2$`WEIR_AND_COCKERHAM_FST`) # lowest value there is 0.248619 (1% cutoff)
min_one_cutoff_M <- min(Fst_M_sort_cut2$`WEIR_AND_COCKERHAM_FST`)
table(Fst_M_sort_cut2$scaf)
quantile(Fst_M$`WEIR_AND_COCKERHAM_FST`, c(0.95, 0.99), na.rm = T)
mean(Fst_M$WEIR_AND_COCKERHAM_FST)
Fst_male <- left_join(scaffold_lengths_M, Fst_M) %>% filter(WEIR_AND_COCKERHAM_FST>=0)

#for x axis, we want cumulative bases for each position in the genome for a continuous axis
nCHR_M <- length(unique(Fst_male$scaf))
Fst_male$BPcum <- 0
sm <- 0
nbp_M <- c()
for (i in unique(Fst_male$scaf)){
  nbp_M[i] <- max(Fst_male[Fst_male$scaf == i,]$base)
  Fst_male[Fst_male$scaf == i,"BPcum"] <- Fst_male[Fst_male$scaf == i,"base"] + sm
  sm <- sm + nbp_M[i]
}
axis.set_M <- Fst_male %>%
  group_by(scaf) %>%
  summarize(center = (max(BPcum) + min(BPcum)) / 2)

Fst_male %>% ggplot(aes(x=BPcum, y=WEIR_AND_COCKERHAM_FST, color=as.factor(scaf))) +
  geom_point(alpha = 0.75) +
  scale_x_continuous(label = axis.set_M$scaf, breaks = axis.set_M$center) +
  theme(axis.text.x=element_text(angle = 90), legend.position="none") +
  labs(x="Chromosome", y="Weir & Cockerham Fst", color="") +
  #guides(color = guide_legend(ncol = 3, byrow = F)) +
  geom_hline(yintercept = min_five_cutoff_M, linetype="dotted", color = "black", size=0.75) +
  geom_hline(yintercept = min_one_cutoff_M, linetype="dotted", color = "black", size=1.5)  +
  #geom_vline(xintercept = as.integer(sm), linetype="dotted", color="red",size=1) +
  labs(title="Fst Male Reference - no filters\n (5% and 1% cutoff as dotted lines)")
ggsave("visualizations/Figure_Step2-Fst_maleRef_manhattan_with95_99_noMissing_all.pdf")
```

##### No Outlier
```{r fst_maleRef_no_outliers_lg, echo=FALSE}
Fst_M <- read_tsv("./step2-fst/subset_analyses/maleRef/maleRef_no_outliers_pruned.weir.fst") %>%
  replace_na(list(WEIR_AND_COCKERHAM_FST=0)) %>% 
  rename(scaf = CHROM, base=POS) %>% mutate(base=as.numeric(base)) %>% 
  filter(WEIR_AND_COCKERHAM_FST >=0) %>% 
  replace_na(list(WEIR_AND_COCKERHAM_FST=0))
scaffold_lengths_M <- read_tsv("./step0-variantcalling/maleRef/CHRR_integrated.fa.fai", col_names = c("scaf","length")) %>% filter(grepl("lg",scaf))

cutoff1_M <- round(nrow(Fst_M)*0.05)
Fst_M_sort_cut1 <- head(Fst_M %>% arrange(-WEIR_AND_COCKERHAM_FST), n=cutoff1_M)
min(Fst_M_sort_cut1$WEIR_AND_COCKERHAM_FST) # lowest value there is 0.120286 (5% cutoff)
min_five_cutoff_M <- min(Fst_M_sort_cut1$WEIR_AND_COCKERHAM_FST)

cutoff2_M <- round(nrow(Fst_M)*0.01)
Fst_M_sort_cut2 <- head(Fst_M %>% arrange(-WEIR_AND_COCKERHAM_FST), n=cutoff2_M)
min(Fst_M_sort_cut2$WEIR_AND_COCKERHAM_FST) # lowest value there is 0.186137 (1% cutoff)
min_one_cutoff_M <- min(Fst_M_sort_cut2$WEIR_AND_COCKERHAM_FST)

table(Fst_M_sort_cut2$scaf)
quantile(Fst_M$WEIR_AND_COCKERHAM_FST, c(0.95, 0.99), na.rm = T)
mean(Fst_M$WEIR_AND_COCKERHAM_FST) # 0.03747152
Fst_male <- left_join(scaffold_lengths_M, Fst_M) %>% filter(WEIR_AND_COCKERHAM_FST>=0)

#for x axis, we want cumulative bases for each position in the genome for a continuous axis
nCHR_M <- length(unique(Fst_male$scaf))
Fst_male$BPcum <- 0
sm <- 0
nbp_M <- c()
for (i in unique(Fst_male$scaf)){
  nbp_M[i] <- max(Fst_male[Fst_male$scaf == i,]$base)
  Fst_male[Fst_male$scaf == i,"BPcum"] <- Fst_male[Fst_male$scaf == i,"base"] + sm
  sm <- sm + nbp_M[i]
}
axis.set_M <- Fst_male %>%
  group_by(scaf) %>%
  summarize(center = (max(BPcum) + min(BPcum)) / 2)
Fst_male %>% ggplot(aes(x=BPcum, y=WEIR_AND_COCKERHAM_FST, color=as.factor(scaf))) +
  geom_point(alpha = 0.7) +
  scale_x_continuous(label = axis.set_M$scaf, breaks = axis.set_M$center) +
  theme(axis.text.x = element_text(angle = 90)) +
  labs(x="Chomosome", y="Weir & Cockerham Fst", color="") +
  guides(color=guide_legend(ncol=3, byrow=F)) +
  geom_hline(yintercept=min_five_cutoff_M, linetype="dotted", color = "black", size=0.75) +
  geom_hline(yintercept=min_one_cutoff_M, linetype="dotted", color = "black", size=1.5)  +
  #geom_vline(xintercept = as.integer(sm), linetype="dotted", color="red",size=1) +
  labs(title="Fst Male Reference - filtered dataset\n (5% and 1% cutoff as dotted lines)")
ggsave("visualizations/Figure_Step2-Fst_maleRef_manhattan_with95_99_noMissing_no_outliers.pdf")
````

```{r fst_maleRef_no_outliers_scaffold, echo=FALSE}
Fst_M <- read_tsv("./step2-fst/subset_analyses/maleRef/maleRef_no_outliers_pruned.weir.fst") %>%
  replace_na(list(WEIR_AND_COCKERHAM_FST=0)) %>% 
  rename(scaf = CHROM, base=POS) %>% mutate(base=as.numeric(base)) %>% 
  filter(WEIR_AND_COCKERHAM_FST >=0) %>% 
  replace_na(list(WEIR_AND_COCKERHAM_FST=0))
scaffold_lengths_M <- read_tsv("./step0-variantcalling/maleRef/CHRR_integrated.fa.fai", col_names = c("scaf","length")) %>% filter(grepl("scaffold",scaf))

cutoff1_M <- round(nrow(Fst_M)*0.05)
Fst_M_sort_cut1 <- head(Fst_M %>% arrange(-WEIR_AND_COCKERHAM_FST), n=cutoff1_M)
min(Fst_M_sort_cut1$WEIR_AND_COCKERHAM_FST) # lowest value there is 0.120286 (5% cutoff)
min_five_cutoff_M <- min(Fst_M_sort_cut1$WEIR_AND_COCKERHAM_FST)

cutoff2_M <- round(nrow(Fst_M)*0.01)
Fst_M_sort_cut2 <- head(Fst_M %>% arrange(-WEIR_AND_COCKERHAM_FST), n=cutoff2_M)
min(Fst_M_sort_cut2$WEIR_AND_COCKERHAM_FST) # lowest value there is 0.186137 (1% cutoff)
min_one_cutoff_M <- min(Fst_M_sort_cut2$WEIR_AND_COCKERHAM_FST)

table(Fst_M_sort_cut2$scaf)
quantile(Fst_M$WEIR_AND_COCKERHAM_FST, c(0.95, 0.99), na.rm = T)
mean(Fst_M$WEIR_AND_COCKERHAM_FST) # 0.03747152
Fst_male <- left_join(scaffold_lengths_M, Fst_M) %>% filter(WEIR_AND_COCKERHAM_FST>=0)

#for x axis, we want cumulative bases for each position in the genome for a continuous axis
nCHR_M <- length(unique(Fst_male$scaf))
Fst_male$BPcum <- 0
sm <- 0
nbp_M <- c()
for (i in unique(Fst_male$scaf)){
  nbp_M[i] <- max(Fst_male[Fst_male$scaf == i,]$base)
  Fst_male[Fst_male$scaf == i,"BPcum"] <- Fst_male[Fst_male$scaf == i,"base"] + sm
  sm <- sm + nbp_M[i]
}
axis.set_M <- Fst_male %>%
  group_by(scaf) %>%
  summarize(center = (max(BPcum) + min(BPcum)) / 2)
Fst_male %>% ggplot(aes(x=BPcum, y=WEIR_AND_COCKERHAM_FST, color=as.factor(scaf))) +
  geom_point(alpha = 0.7) +
  scale_x_continuous(label = axis.set_M$scaf, breaks = axis.set_M$center) +
  theme(axis.text.x = element_text(angle = 90), legend.position="none") +
  labs(x="Chomosome", y="Weir & Cockerham Fst", color="") +
  #guides(color=guide_legend(ncol=3, byrow=F)) +
  geom_hline(yintercept=min_five_cutoff_M, linetype="dotted", color = "black", size=0.75) +
  geom_hline(yintercept=min_one_cutoff_M, linetype="dotted", color = "black", size=1.5)  +
  #geom_vline(xintercept = as.integer(sm), linetype="dotted", color="red",size=1) +
  labs(title="Fst Male Reference - filtered dataset\n (5% and 1% cutoff as dotted lines)")
ggsave("visualizations/Figure_Step2-Fst_maleRef_manhattan_with95_99_noMissing_no_outliers_scaffolds.pdf")
````

## Step 2: kmersgwas
### All Individuals
### No Outliers
