# Delta Smelt Genome Sample Tracking

[toc]


## Directories

### Local

`miocene/Desktop/work/smelt/genome`
`miocene/Desktop/git_repos`


## Emails

### 2017

* **12.07** = mtg with Matt/Ruta/Lutz

### 2018
* **07.18** = email from Shannon inquiry to do PacBio & Hi-C (outsourced)
* **07.20** = email to perhaps to 10X bc of small size of smelt
* **08.03** = mtg to discuss next steps 
* **09.18** = email from shannon saying will sample fish and drop off tissue on 09/25 ("Delta Smelt Sequencing Sample Prep")
* **09.25** = sample drop off (shannon --> DNA core)
* **09.27** = email from shannon with sample sheet
* **10.11** = email from ruta with photo of gel "These two samples look suitable for 10X genomics."
* **10.16** = email from ruta "Thanks for the phone call. As we discussed, we will proceed with 10X library prep for DNA extracted from back muscle tissue of Female 2."
* **10.31** = email from shannon with 10X sample submission form requesting 120x coverage of _**female 02**_

### 2019
* **03.31** = email from shannon to ruta and diana requesting methods details about _**female 02**_ extraction and 10x prep
* **04.02** = email from ruta w reply (see below)






## Female 02

### Tissue Sampling

* date: 2018.09.27
* types:
	* muscle
	* heart
	* organs
* sample sheet: 
	- `miocene/Desktop/work/smelt/genome/2018.10_DeltaSmelt_HMWgDNAextraction`
	- https://ucdavis.coreomics.com/submissions/2be9dca14593



### DNA Extraction

1. Do you follow a published HMW extraction protocol I can cite?
	I used the protocol published in this paper  https://www.ncbi.nlm.nih.gov/pubmed/14641478
2. Do you follow a publish 10X library prep I can cite?
	I having copied Diana, our 10X prep expert who can provide you information about the 10X lib prep.
3. Do you know how much HMW DNA was extracted from the female used for sequencing?
4. Do you know the concentration of the HMW DNA? (this question may be answered from the above question but just trying to cover all my bases!)
	We extracted ~3.4µg of DNA at 87ng/µl concentration. That said, it is sometimes difficult to quantify the HMW DNA accurately and these concentration may be a little off.

### 10X

from diana - female - sent April 02, 2019 - eml:
	Genomic DNA was adjusted to a concentration of `0.91 ng/µl` and `1.14 ng` of template gDNA was loaded on a Chromium Genome Chip. Whole genome sequencing libraries were prepared using Chromium Genome Library & Gel Bead Kit v.2 (10X Genomics, cat. 120258), Chromium Genome Chip Kit v.2 (10X Genomics, cat. 120257), Chromium i7 Multiplex Kit (10X Genomics, cat. 120262) and Chromium controller according to manufacturer’s instructions with one modification. Briefly, gDNA was combined with Master Mix, a library of Genome Gel Beads, and partitioning oil to create Gel Bead-in-Emulsions (GEMs) on a Chromium Genome Chip. The GEMs were isothermally amplified with primers containing an Illumina Read 1 sequencing primer, a unique 16-bp 10x bar-code and a 6-bp random primer sequence, and bar-coded DNA fragments were recovered for Illumina library construction. The amount and fragment size of post-GEM DNA was quantified prior using a Bioanalyzer 2100 with an Agilent High sensitivity DNA kit (Agilent, cat. 5067-4626). Prior to Illumina library construction, the GEM amplification product was sheared on an E220 Focused Ultrasonicator (Covaris, Woburn, MA) to approximately 350bp `(50 seconds at peak power = 175, duty factor = 10, and cycle/burst = 200)`. Then, the sheared GEMs were converted to a sequencing library following the 10X standard operating procedure. The library was quantified by qPCR with a Kapa Library Quant kit (Kapa Biosystems-Roche) and sequenced on NovaSeq6000 sequencer (Illumina, San Diego, CA) with paired-end 150 bp reads.

from diana - male - send July 04, 2019 eml:"Comparing DNA Starting Quantities" 
	Genomic DNA was adjusted to a concentration of `0.9 ng/µl` and `1.12 ng` of template gDNA was loaded on a Chromium Genome Chip. Whole genome sequencing libraries were prepared using Chromium Genome Library & Gel Bead Kit v.2 (10X Genomics, cat. 120258), Chromium Genome Chip Kit v.2 (10X Genomics, cat. 120257), Chromium i7 Multiplex Kit (10X Genomics, cat. 120262) and Chromium controller according to manufacturer’s instructions with one modification. Briefly, gDNA was combined with Master Mix, a library of Genome Gel Beads, and partitioning oil to create Gel Bead-in-Emulsions (GEMs) on a Chromium Genome Chip. The GEMs were isothermally amplified with primers containing an Illumina Read 1 sequencing primer, a unique 16-bp 10x bar-code and a 6-bp random primer sequence, and bar-coded DNA fragments were recovered for Illumina library construction. The amount and fragment size of post-GEM DNA was quantified prior using a Bioanalyzer 2100 with an Agilent High sensitivity DNA kit (Agilent, cat. 5067-4626). Prior to Illumina library construction, the GEM amplification product was sheared on an E220 Focused Ultrasonicator (Covaris, Woburn, MA) to approximately 350bp `(55 seconds at peak power = 175, duty factor = 10, and cycle/burst = 200)`. Then, the sheared GEMs were converted to a sequencing library following the 10X standard operating procedure. The library was quantified by qPCR with a Kapa Library Quant kit (Kapa Biosystems-Roche) and sequenced on one shared lane of NovaSeq S4 flow cell (Illumina, San Diego, CA) with paired-end 150 bp reads.



## Submission forms

* 2019/08/07 [HMW Extraction](https://ucdavis.coreomics.com/submissions/776423c74a37)
	* Female 02
* 2019/11/07 [PacBio & 10X](https://ucdavis.coreomics.com/submissions/2be9dca14593)
	* **NEW** female for PacBio
	* Request male 10X, PacBio and hi-c (hi-c to be outsourced)
	* PG and isolated
* 2019/12/04 [10X](https://ucdavis.coreomics.com/submissions/979394057d81)
	* Male 02




