\documentclass[12pt,a4paper]{article}
\begin{document}
\begin{table}[ht]
\begin{center}
\caption{All statistics are based on contigs of size $\geq$ 3000 bp, unless otherwise noted (e.g., "\# contigs ($\geq$ 0 bp)" and "Total length ($\geq$ 0 bp)" include all contigs).}
\begin{tabular}{|l*{84}{|r}|}
\hline
Assembly & \# contigs ($\geq$ 0 bp) & \# contigs ($\geq$ 1000 bp) & \# contigs ($\geq$ 5000 bp) & \# contigs ($\geq$ 10000 bp) & \# contigs ($\geq$ 25000 bp) & \# contigs ($\geq$ 50000 bp) & Total length ($\geq$ 0 bp) & Total length ($\geq$ 1000 bp) & Total length ($\geq$ 5000 bp) & Total length ($\geq$ 10000 bp) & Total length ($\geq$ 25000 bp) & Total length ($\geq$ 50000 bp) & \# contigs & Largest contig & Total length & Reference length & Estimated reference length & GC (\%) & Reference GC (\%) & N50 & NG50 & N90 & NG90 & auN & auNG & L50 & LG50 & L90 & LG90 & \# total reads & \# left & \# right & Mapped (\%) & Reference mapped (\%) & Properly paired (\%) & Reference properly paired (\%) & Avg. coverage depth & Reference avg. coverage depth & Coverage $\geq$ 1x (\%) & Reference coverage $\geq$ 1x (\%) & \# large blocks misassemblies & \# misassemblies & \# misassembled contigs & Misassembled contigs length & \# local misassemblies & \# scaffold gap ext. mis. & \# scaffold gap loc. mis. & \# structural variations & \# possible TEs & \# unaligned mis. contigs & \# unaligned contigs & Unaligned length & Genome fraction (\%) & Duplication ratio & Avg contig read support & \# N's per 100 kbp & \# mismatches per 100 kbp & \# indels per 100 kbp & \# genomic features & \# operons & Complete BUSCO (\%) & Partial BUSCO (\%) & \# predicted genes (unique) & \# predicted genes ($\geq$ 0 bp) & \# predicted genes ($\geq$ 300 bp) & \# predicted genes ($\geq$ 1500 bp) & \# predicted genes ($\geq$ 3000 bp) & \# predicted rRNA genes & Largest alignment & Total aligned length & NA50 & NGA50 & NA90 & NGA90 & auNA & auNGA & LA50 & LGA50 & LA90 & LGA90 & K-mer-based compl. (\%) & K-mer-based cor. length (\%) & K-mer-based mis. length (\%) & \# k-mer-based misjoins \\ \hline
ipa & 1805 & 1805 & 1805 & 1805 & 1800 & 1773 & 436920153 & 436920153 & 436920153 & 436920153 & 436829068 & 435805794 & 1805 & 3341941 & 436920153 & - & 490000000 & 44.99 & - & 418614 & 346609 & 97153 & - & 668619.6 & 596190.5 & 264 & 334 & 1171 & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & 0.07 & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - \\ \hline
chromonomer & 376 & 376 & 376 & 376 & 372 & 361 & 437273953 & 437273953 & 437273953 & 437273953 & 437196775 & 436784290 & 376 & 22675904 & 437273953 & - & 490000000 & 44.99 & - & 14850352 & 14090550 & 583287 & - & 12187194.5 & 10875801.4 & 13 & 15 & 61 & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & 80.98 & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - & - \\ \hline
\end{tabular}
\end{center}
\end{table}
\end{document}
